pymatgen/io/vasp/inputs.py

```diff
@@ -2674,14 +2674,14 @@ def from_directory(cls, input_dir: str, optional_files: dict | None = None) -> S
         ]:
             try:
                 full_zpath = zpath(os.path.join(input_dir, fname))
-                sub_dct[fname.lower()] = ftype.from_file(full_zpath)
+                sub_dct[fname.lower()] = ftype.from_file(full_zpath)  # type: ignore[attr-defined]
             except FileNotFoundError:  # handle the case where there is no KPOINTS file
                 sub_dct[fname.lower()] = None
 
         sub_dct["optional_files"] = {}
         if optional_files is not None:
             for fname, ftype in optional_files.items():
-                sub_dct["optional_files"][fname] = ftype.from_file(os.path.join(input_dir, fname))
+                sub_dct["optional_files"][fname] = ftype.from_file(os.path.join(input_dir, fname))  # type: ignore[attr-defined]
```

Comment:
Wondering if there is a better solution here

Reply:
my first though was to define a protocol

```py
from typing import Protocol

class FromFileable(Protocol):
    @classmethod
    def from_file(cls, filename: str) -> Self: ...
```

looks like this solves the problem for `pylance` (a different slightly better type checker from microsoft) but `mypy` doesn't seem to support this

Reply:
i don't think there's a better solution

Reply:
Thanks. But that `Protocol` is certainly a good tool to have!